shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1396,12 +1504,28 @@ - (void)setupKeyboardAnimationVsyncClient {
       // Ensure the keyboardAnimationView is in view hierarchy when animation running.
       [flutterViewController.get().view addSubview:[flutterViewController keyboardAnimationView]];
     }
-    if ([flutterViewController keyboardAnimationView].layer.presentationLayer) {
-      CGFloat value =
-          [flutterViewController keyboardAnimationView].layer.presentationLayer.frame.origin.y;
-      flutterViewController.get()->_viewportMetrics.physical_view_inset_bottom = value;
-      [flutterViewController updateViewportMetrics];
-    }
+
+    KeyboardAnimationView* keyboardView =
+        (KeyboardAnimationView*)[flutterViewController keyboardAnimationView];
+
+    fml::TimeDelta timeElapsed = recorder.get()->GetVsyncTargetTime() - keyboardView.startTime;
+    double frameRate = [[flutterViewController keyboardAnimationVSyncClient] getRefreshRate];
+
+    double animationSettlingDuration = [flutterViewController keyboardSpringCurve].settlingDuration;
```

Comment:
Is This algorithm right？Calculate the stops using this
```objc
[DisplayLinkManager displayRefreshRate];
``` 
but here using 
```objc
double frameRate = [[flutterViewController keyboardAnimationVSyncClient] getRefreshRate];
```
When `frameRate` is not the same with `[DisplayLinkManager displayRefreshRate]`, this algorithm will be wrong.

Reply:
I'm not quite sure on this, as I'm not very familiar with the API. When i call the DisplayLinkManager refreshRate, there isn't an instance of the keyboardAnimationVSyncClient. Is there a better way to access the device's max refreshRate from the engine?

Really I just want the device's max refresh rate - the algorithm won't be affected if the vsync refresh rate is lower (it'll just skip more stops)

Reply:
The first one is the device's max frame rate, but the second is current frame rate. It is not the same^_^

Reply:
Okay good, that's what I wanted!

The first time is called to populate the stops array to support up to the device's maximum frame rate (say 120fps).

The second call is for the current frame rate (which can be variable). If the frame rate is 120fps, great, it will use all the calculated stops. If it's less than 120, say 60, it will skip and use every other stop. It approximates the nearest calculated stop to use with the `frameApproximation` value.

Reply:
But `expectedFrames` is the frame in max frame rate not current frame rate. I think this is still not correct. I didn't locally test your code. But I think the behavior will not be correct. 

You can test using a 120FPS device like iPhone13 Pro or Iphone 14 Pro, and turn on the low power mode(the current frame rate will be 60FPS), and test this is correct or not😄.

Reply:
:) nice trick, didn't know about that

i reworked the expected frames to use the FrameTimingsRecorder to calculate current frame rate, works great now!